--- origsrc/bakefile-0.2.3/autoconf/bakefile.m4	2008-01-06 08:43:23.000000000 -0600
+++ src/bakefile-0.2.3/autoconf/bakefile.m4	2008-03-23 16:47:51.437500000 -0500
@@ -423,8 +423,8 @@
 
       *-*-cygwin* | *-*-mingw32* )
         PIC_FLAG=""
-        SHARED_LD_CC="\$(CC) -shared -o"
-        SHARED_LD_CXX="\$(CXX) -shared -o"
+        SHARED_LD_CC="\$(CC) -shared -Wl,--enable-auto-image-base -o"
+        SHARED_LD_CXX="\$(CXX) -shared -Wl,--enable-auto-image-base -o"
         WINDOWS_IMPLIB=1
       ;;
 
@@ -658,7 +658,7 @@
 
     LDFLAGS_GUI=
     case ${BAKEFILE_HOST} in
-        *-*-cygwin* | *-*-mingw32* )
+        *-*-mingw32* )
         LDFLAGS_GUI="-mwindows"
     esac
     AC_SUBST(LDFLAGS_GUI)
